### PR TITLE
[#287] Add Java linting

### LINKED ISSUE #287: Add Java linting
We want our Java code to be linted, preferably with [check style](https://checkstyle.sourceforge.io/)

### PR DESCRIPTION
**Pull request by chrismatix**
fixes #287

### CODE REVIEW

File: backend/api/admin/src/test/java/co/airy/core/api/admin/ChannelsControllerTest.java [hunk 1]
```diff
@@ -76,8 +76,8 @@ static void afterAll() throws Exception {
 
     private static boolean testDataInitialized = false;
 
-    final static String facebookToken = "token";
-    final static Channel connectedChannel = Channel.newBuilder()
+    static final String facebookToken = "token";
```
Comment (chrismatix) [inline backend/api/admin/src/test/java/co/airy/core/api/admin/ChannelsControllerTest.java:79]: TIL that there is a suggested order for these in the Java spec

File: backend/api/communication/src/test/java/co/airy/core/api/communication/util/TestConversation.java [hunk 2]
```diff
@@ -55,6 +55,10 @@ public static TestConversation from(String conversationId, Channel channel, Map<
         return testConversation;
     }
 
+    public static List<ProducerRecord<String, SpecificRecordBase>> generateRecords(String conversationId, Channel channel, int messageCount) {
```
Comment (chrismatix) [inline backend/api/communication/src/test/java/co/airy/core/api/communication/util/TestConversation.java:58]: overloaded methods should be grouped together

File: backend/lib/kafka/streams/src/main/java/co/airy/kafka/streams/KafkaStreamsWrapper.java [hunk 3]
```diff
@@ -175,11 +176,11 @@ public KafkaStreamsWrapper withExcludeInternalTopics(boolean excludeInternalTopi
         int retries = 0;
         do
             try {
-                return streams.store(storeName, QueryableStoreTypes.keyValueStore());
+                return streams.store(StoreQueryParameters.fromNameAndType(storeName, QueryableStoreTypes.keyValueStore()));
```
Comment (chrismatix) [inline backend/lib/kafka/streams/src/main/java/co/airy/kafka/streams/KafkaStreamsWrapper.java:179]: Fixed a deprecation

File: backend/lib/uuid/BUILD [hunk 4]
```diff
@@ -1,5 +1,7 @@
 load("@rules_java//java:defs.bzl", "java_library")
 
+# Due to a bug in the abbreviation checkstyle rule
```
Comment (chrismatix) [inline backend/lib/uuid/BUILD:3]: For some reason I cannot add `UUID` or `UUIDv5` to the list of ignored abbreviations, so this is a workaround

File: checkstyle.xml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        <module name="MethodTypeParameterName">
+            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
+            <message key="name.invalidPattern"
+                     value="Method type name ''{0}'' must match pattern ''{1}''."/>
+        </module>
+        <module name="InterfaceTypeParameterName">
+            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
+            <message key="name.invalidPattern"
+                     value="Interface type name ''{0}'' must match pattern ''{1}''."/>
+        </module>
```
Comment (chrismatix) [inline checkstyle.xml:107]: A lot of these are common sense conventions that we never violate anyway, so it makes sense to keep them.

File: backend/api/admin/src/test/java/co/airy/core/api/admin/ChannelsControllerTest.java (hunk 1 above)
Comment (lucapette) [reply backend/api/admin/src/test/java/co/airy/core/api/admin/ChannelsControllerTest.java:79]: yes!

File: .bazelrc [hunk 6]
```diff
@@ -7,6 +7,9 @@ build --java_toolchain=@bazel_tools//tools/jdk:toolchain_java11 --host_java_tool
 
 # Output test errors by default
 test --test_output=errors
+# Do not build production artifacts to speed up tests
+# Build outputs should be validated by running build
+test --build_tests_only
```
Comment (chrismatix) [inline .bazelrc:12]: I think this is also a common sense setting for developers

**Review by paulodiniz — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -35,6 +35,33 @@ life meaning" and its number is 4242, the branch name will be
 
 You can push as many commits we need to finish a feature.
 
+## Lint your changes
+
+We use a couple of language specific linters. For Bazel code we use [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier),
+for web projects [prettier](https://prettier.io/), and for Java [CheckStyle](https://checkstyle.sourceforge.io/).
```
Comment (lucapette) [inline CONTRIBUTING.md:41]:
```suggestion
We use language specific linters: [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) to lint Bazel files, [prettier](https://prettier.io/) for TypeScript, JavaScript and SCSS, [CheckStyle](https://checkstyle.sourceforge.io/) for Java.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -35,6 +35,33 @@ life meaning" and its number is 4242, the branch name will be
...
+
+Java linting is run as a test, so depending on the package you can run
```
Comment (lucapette) [inline CONTRIBUTING.md:43]:
```suggestion
Java linting is run as a test, so depending on the package you can run:
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -35,6 +35,33 @@ life meaning" and its number is 4242, the branch name will be
...
+
+```sh
+bazel test //my/package:checkstyle
+```
+
+to have Bazel check your Java code for linting errors.
+
+Buildifier and Prettier are bundled up in a single Bazel command. So you can run
```
Comment (lucapette) [inline CONTRIBUTING.md:51]:
```suggestion
Buildifier and Prettier are bundled up in a single Bazel command. So you can run:
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -35,6 +35,33 @@ life meaning" and its number is 4242, the branch name will be
...
+
+```sh
+bazel run //:check
+```
+
+to check your code and
```
Comment (lucapette) [inline CONTRIBUTING.md:57]:
```suggestion
to check your code and:
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -35,6 +35,33 @@ life meaning" and its number is 4242, the branch name will be
...
+
+```sh
+bazel run //:fix
+```
+
+to have the linters try and fix and the issues.
```
Comment (lucapette) [inline CONTRIBUTING.md:63]:
```suggestion
to try fixing linting issues automatically.
```

**Review by lucapette — CHANGES_REQUESTED** [review]
Requesting changes so we can improve the writing of the docs (I left suggestions so we can simplify the process). Can't wait to merge!

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**